### PR TITLE
testsuite: add multiple key job-info lookup tests

### PR DESCRIPTION
**Pull request by chu11**
Problem: In PR #5520 "multiple key" "flux job info" tests were removed due to a feature change in "flux job info".  This removed some coverage of corner cases in job-info.lookup.

Add a new "job-info.lookup" RPC target helper tool and re-add a number of the previously removed "multiple key" tests.

### CODE REVIEW

File: t/job-info/info_lookup.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    if (!(h = flux_open (NULL, 0)))
+        log_err_exit ("flux_open");
+
+    if (flux_job_id_parse (argv[1], &id) < 0)
+        log_msg_exit ("error parsing jobid: %s", argv[1]);
+
+    if (!(keys = json_array ()))
+        log_err_exit ("json_array");
```
Comment (garlick) [inline t/job-info/info_lookup.c:41]: jansson functions don't claim to set errno so this should probably use `log_msg_exit()`.

File: t/job-info/info_lookup.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    for (i = 2; i < argc; i++) {
+        json_t *key = json_string (argv[i]);
+        if (!key)
+            log_err_exit ("json_string_value");
+        if (json_array_append_new (keys, key) < 0)
+            log_err_exit ("json_array_append_new");
```
Comment (garlick) [inline t/job-info/info_lookup.c:48]: same error handling comment

**Review by garlick — APPROVED** [review]
LGTM, just a trivial change requested to error logging.

**Comment by chu11** [comment]
thanks!  tweaked and setting MWP
